forage-android/src/main/java/com/joinforage/forage/android/network/model/Balance.kt

```diff
@@ -0,0 +1,40 @@
+package com.joinforage.forage.android.network.model
+
+import org.json.JSONObject
+
+/**
+ * A model that represents the available balance on a customer’s card.
+ * @see [EbtBalance]
+ */
+interface Balance
+
+/**
+ * @param snap The available SNAP balance on the customer’s EBT Card, represented as a numeric string.
+ * @param cash The available EBT Cash balance on the customer’s EBT Card, represented as a numeric string.
+ */
+data class EbtBalance(
+    val snap: String,
+    val cash: String
+) : Balance {
+    constructor(jsonObject: JSONObject) : this(
+        jsonObject.getString("snap"),
+        jsonObject.getString("non_snap")
+    )
+
+    internal companion object {
+        /** ex: { "snap": "10.00", "cash": "10.00" } */
+        internal fun fromSdkResponse(jsonString: String): EbtBalance {
+            val jsonObject = JSONObject(jsonString)
+            val snap = jsonObject.getString("snap")
+            val cash = jsonObject.getString("cash")
+            return EbtBalance(
+                snap = snap,
+                cash = cash
+            )
+        }
+    }
```

Comment:
```suggestion
constructor(jsonString: String) : this(JSONObject(jsonString)
```
Thoughts on getting rid of the companion object entirely with another constructor `jsonString: String`?

Reply:
@devinmorgan 

```kotlin
// the JSON object constructor is not compatible with the "SDK response"
 constructor(jsonObject: JSONObject) : this(
        jsonObject.getString("snap"),
        jsonObject.getString("non_snap")
    )
```

The other (`Payment`, `PaymentMethod`, ...) constructors map the raw/unprocessed JSON response from the Payments API.

Whereas the Balance's `non_snap` field is already mapped/renamed to `cash`, which is why I need a custom `fromSdkResponse` constructor – to make it explicitly clear that the factory is only compatible with the "SDK response" as input.

Alternatively, I can make a constructor like `cash = json.get('non_snap') ?: json.get('cash')` – that is compatible with either format.

What are your thoughts on the alternative?

Reply:
In any case, this code will likely get simplified when we introduce _actual_ typed SDK responses (instead of JSON strings)

Reply:
cool sounds like it's no a quick win so let's skip for now! Thanks for considering!